swssconfig/swssconfig.cpp

```diff
@@ -127,47 +134,88 @@ bool load_json_db_data(iostream &fs, vector<KeyOpFieldsValuesTuple> &db_items)
     return true;
 }
 
+vector<string> read_directory(const string &path)
+{
+    vector<string> ret;
+    DIR *dir;
+    dirent *entry;
+    dir = opendir(path.c_str());
+    if (dir)
+    {
+        while (true)
+        {
+            entry = readdir(dir);
+            if (entry == NULL)
+            {
+                break;
+            }
+            if (strcmp(entry->d_name, ".") && strcmp(entry->d_name, ".."))
+            {
+                ret.push_back(string(entry->d_name));
+            }
+        }
+        closedir(dir);
+    }
+    sort(ret.begin(), ret.end());
+    return ret;
+}
+
 int main(int argc, char **argv)
 {
-    Logger::setMinPrio(Logger::SWSS_DEBUG);
+    Logger::setMinPrio(Logger::SWSS_NOTICE);
 
-    if (argc != 2)
+    vector<string> files;
+    if (argc == 1)
     {
-        usage(argv);
-        exit(EXIT_FAILURE);
+        files = read_directory(SWSS_CONFIG_DIR);
     }
-
-    vector<KeyOpFieldsValuesTuple> db_items;
-    fstream fs;
-    try
+    if (argc == 2 && !strcmp(argv[1], "-h"))
     {
-        fs.open(argv[1], fstream::in);
-        if (!load_json_db_data(fs, db_items))
-        {
-            SWSS_LOG_ERROR("Failed loading data from json file.");
-            fs.close();
-            return EXIT_FAILURE;
-        }
-        fs.close();
+        usage();
+        exit(EXIT_SUCCESS);
     }
-    catch(...)
+    else
     {
-        cout << "Failed loading json file: " << argv[1] << " Please refer to logs\n";
-        return EXIT_FAILURE;
+        for (auto i = 1; i < argc; i++)
+        {
+            files.push_back(string(argv[i]));
+        }
     }
 
-    try
+    for (auto i : files)
     {
-        if (!write_db_data(db_items))
+        SWSS_LOG_NOTICE("Loading config from JSON file:%s...", i.c_str());
+
+        vector<KeyOpFieldsValuesTuple> db_items;
+        fstream fs;
```

Comment:
I mean moving fs variable inside the loop block, not reuse it. ref: http://stackoverflow.com/questions/2284775/c-can-i-reuse-fstream-to-open-and-write-multiple-files